zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/CamundaExporterIT.java

```diff
@@ -382,4 +334,158 @@ private ExporterResourceProvider mockResourceProvider(
+                            ImmutableProcessInstanceRecordValue.builder()
+                                .from(factory.generateObject(ProcessInstanceRecordValue.class))
+                                .withParentProcessInstanceKey(-1L)
+                                .withBpmnElementType(elementType)
+                                .build();
+                        return factory.generateRecord(
+                            ValueType.PROCESS_INSTANCE,
+                            r ->
+                                r.withIntent(ELEMENT_ACTIVATING)
+                                    .withValue((T) processInstanceRecordValue)
+                                    .withTimestamp(System.currentTimeMillis()));
+                      });
+
+      final Map<Class<?>, Supplier<Record<T>>> customRecordGenerators =
+          Map.of(
+              EventFromJobHandler.class,
+              getJobRecord,
+              ListViewFlowNodeFromJobHandler.class,
+              getJobRecord,
+              ListViewProcessInstanceFromProcessInstanceHandler.class,
+              getProcessInstanceRecord.apply(BpmnElementType.PROCESS),
+              ListViewFlowNodeFromProcessInstanceHandler.class,
+              getProcessInstanceRecord.apply(BpmnElementType.BOUNDARY_EVENT),
+              DecisionEvaluationHandler.class,
+              getDecisionEvalRecord,
+              MetricFromProcessInstanceHandler.class,
+              getProcessInstanceRecord.apply(BpmnElementType.PROCESS),
+              UserTaskProcessInstanceHandler.class,
+              getProcessInstanceRecord.apply(BpmnElementType.PROCESS));
+
+      final Supplier<Record<T>> defaultRecordGen =
+          () ->
+              recordGenerator(
+                  handler,
+                  () ->
+                      factory.generateRecord(
+                          handler.getHandledValueType(),
+                          r -> r.withTimestamp(System.currentTimeMillis())));
+      return () -> customRecordGenerators.getOrDefault(handler.getClass(), defaultRecordGen).get();
+    }
+
+    private <S extends ExporterEntity<S>, T extends RecordValue> void exportTest(
+        final CamundaExporter exporter,
+        final ExportHandler<S, T> handler,
+        final SearchClientAdapter clientAdapter)
+        throws IOException {
+
+      final var currentTime = OffsetDateTime.now();
+      final var mocked = mockStatic(OffsetDateTime.class);
+      mocked.when(OffsetDateTime::now).thenReturn(currentTime);
+
+      final var record = getHandlerRecord(handler).get();
+      final var entityId = handler.generateIds(record).getFirst();
+      final var expectedEntity = handler.createNewEntity(entityId);
+      handler.updateEntity(record, expectedEntity);
+      exporter.export(record);
+
+      mocked.close();
+
+      final var responseEntity =
+          clientAdapter.get(
+              expectedEntity.getId(), handler.getIndexName(), handler.getEntityType());
```

Comment:
👍🏼 I like that we have a way to verify that the actual and expected state with the entity and handler vs what the exporter does.